core-api-library/library/src/main/java/net/gini/android/core/api/Resource.kt

```diff
@@ -0,0 +1,10 @@
+package net.gini.android.core.api
+
+sealed class Resource<T>(
+    val data: T? = null,
+    val message: String? = null,
+    val errorCode: Int? = null
+) {
+    class Success<T>(data: T) : Resource<T>(data)
+    class Error<T>(message: String, code: Int? = null, data: T? = null) : Resource<T>(data, message, code)
```

Comment:
Please add the response status code, headers and the response body as string.

Reply:
Sorry, Levente, I just now realised that my sentence was not clear about which new field should be a string. 🤦 I only meant the body to be a string. Status code should be an Int and headers a Map.